### ISSUE TITLE
Bug with login using web view - method call in the wrong place

### ISSUE THREAD

**Issue by mvukojicic** [microsoft/aad_b2c_webview#17]
Hi, first thanks for creating this library, great work and it's saving some time for us :)
After I tried to implement the web view in my app, I found issues related to it. I will try to explain it in detail and I hope you will not mind because I want to help you improve this package.

So you have the method onPageFinishedTasks(URL, response); that is being called inside of onPageFinished. That method for checking the URL is called only once when the login screen is initially loaded, and that's it. If I input the credentials for login nothing happens, because onPageFinished is calling onPageFinishedTasks(URL, response) only once, and that's the case when the login screen is being loaded.
Further debugging leads me to the conclusion that the right place to call the method onPageFinishedTasks(URL, response); is inside of navigationDelegate property because that field is being triggered every time the user interacts with the screen and tries to log in.
I tried this solution in my app and it's working perfectly, also I have tried your's and it's not working. If you have any questions or maybe I was not clear enough please text me I am open to help :)

**Comment by mohanajuhi166** [comment]
@mvukojicic we would love if you can contribute this change to the repository. Would you be open to that?

**Comment by mvukojicic** [comment]
@mohanajuhi166  sure np, I'll get into it once I find some free time.

**Comment by baudine-justsolve** [comment]
Hi @mvukojicic ,

Can you please share an example of your set up for the widget and changes you have made. I think I have the same issue but I am unable to determine if it's just my setup. For a mobile flutter app, is the redirect url a deeplink or a normal redirect url?

**Comment by mvukojicic** [comment]
Hello @baudine-justsolve  of course, no problem, here's the sample of the code that I used, I am not using this library, I have created my own flow for communication between the app and Azure. 
So, I am using a web view package because I didn't want to open a browser from the app.

![Screenshot 2023-08-25 at 09 38 00](https://github.com/microsoft/aad_b2c_webview/assets/102806738/8cfffb8e-f250-4d7a-b96a-75f7334dfa04)

![Screenshot 2023-08-25 at 09 38 19](https://github.com/microsoft/aad_b2c_webview/assets/102806738/ee4d9743-9bd2-45d0-8ad2-9d3bef5cbbda)

If you have any questions, I am more than happy to help :)

**Comment by baudine-justsolve** [comment]
Hi, thank you so much. I can confirm I did not have the same issue. I got mine working though using ADB2CEmbedWebView

**Comment by mvukojicic** [comment]
I don't know how the package is working for you guys, My colleague and I tried it and it was not working at all :/  You're welcome

**Comment by baudine-justsolve** [comment]
Hi, we are still in development phase but the curent version on pub.get ( 0.0.49 ) was not working. After I used the latest version from github with the fixes, it is working. I can log in, get the code, authorise, get the token, and it redirects correctly. I can also reset my password so all flows are working great. It seems like the issue I was having had to do with the scopes (which is fixed in another issue recently)

**Comment by mohanajuhi166** [comment]
closing the issue.
